src/physics/euler.cpp

```diff
@@ -175,8 +175,12 @@ ::convert_conservative_to_primitive ( const std::array<real2,nstate> &conservati
     dealii::Tensor<1,dim,real2> vel = compute_velocities<real2>(conservative_soln);
     real2 pressure = compute_pressure<real2>(conservative_soln);
 
-    check_positive_quantity<real2>(density, "density");
-    check_positive_quantity<real2>(pressure, "pressure");
+    if (this->all_parameters->limiter_param.bound_preserving_limiter != Parameters::LimiterParam::LimiterType::positivity_preservingZhang2010
```

Comment:
Adding a change in `handle_non_physical_result` would have cluttered the results quite a bit as a message would be output at every node that had a negative value for density/pressure to indicate that a limiter is being applied to prevent this. So instead of changing `handle_non_physical_result`, I modified `check_positive_quantity` function as suggested. If the positivity preserving limiters are used, the function skips the check. This is part of my latest changes which I will push soon.

Reply:
Looks good! I agree with your reasoning to keep the different behaviour in Euler.